### PR TITLE
change subpar/amp volume to dB

### PR DESCRIPTION
**Pull request by michiboo**
Hi I have change volume to dB in subpar/amp. I have tested myself it seem to be working!

### CODE REVIEW

File: src/Tests/guitar-adnote.xmz [hunk 1]
```diff
@@ -992,7 +992,7 @@ version-revision="4" ZynAddSubFX-author="Nasca Octavian Paul">
 </HARMONICS>
 <AMPLITUDE_PARAMETERS>
 <par_bool name="stereo" value="yes" />
-<par name="volume" value="96" />
+<par_real name="volume" value="0.47" exact_value="0x3EF0A3D7" />
```
Comment (fundamental) [inline src/Tests/guitar-adnote.xmz]: Wrong value based upon some quick tests, it should correspond with +0dB

File: src/Synth/SUBnote.cpp [hunk 2]
```diff
@@ -99,7 +99,7 @@ void SUBnote::setup(float freq,
     this->velocity = velocity;
     portamento  = portamento_;
     NoteEnabled = ON;
-    volume      = powf(0.1f, 3.0f * (1.0f - pars.PVolume / 96.0f)); //-60 dB .. 0 dB
+    volume      = -powf(10.0f,  pars.Volume / 20.0f);
```
Comment (fundamental) [inline src/Synth/SUBnote.cpp]: volume scale factors should be positive, not negative.

File: src/Params/SUBnoteParameters.cpp [hunk 3]
```diff
@@ -38,7 +38,7 @@ static const rtosc::Ports SUBnotePorts = {
     rSelf(SUBnoteParameters),
     rPaste,
     rToggle(Pstereo,    rShort("stereo"), rDefault(true), "Stereo Enable"),
-    rParamZyn(PVolume,  rShort("volume"), rDefault(96), "Volume"),
+    rParamF(Volume,  rShort("volume"), rDefault(0.47f), rLinear(-60.0f,20.0f), "Volume"),
```
Comment (fundamental) [inline src/Params/SUBnoteParameters.cpp]: old eqn 0.1^(3*(1-Pvolume/96)). With an old default of 96 that means 0.1^(3*(1-96/96)) = 0.1^ (3*(1-1)) = 0.1^0 = 1. When a multiplier of one is converted to dB you have 20log_10(1) = +0dB.

File: src/Params/SUBnoteParameters.cpp [hunk 4]
```diff
@@ -270,7 +279,7 @@ float SUBnoteParameters::convertHarmonicMag(int mag, int type)
 
 void SUBnoteParameters::defaults()
 {
-    PVolume  = 96;
+    Volume  = 0.47f;
```
Comment (fundamental) [inline src/Params/SUBnoteParameters.cpp]: wrong default

**Comment by michiboo** [comment]
I have updated required change, it seem it is really working now and passed all the test 
https://travis-ci.org/zynaddsubfx/zynaddsubfx/builds/515025005

File: src/Params/SUBnoteParameters.cpp [hunk 5]
```diff
@@ -38,7 +38,7 @@ static const rtosc::Ports SUBnotePorts = {
     rSelf(SUBnoteParameters),
     rPaste,
     rToggle(Pstereo,    rShort("stereo"), rDefault(true), "Stereo Enable"),
-    rParamZyn(PVolume,  rShort("volume"), rDefault(96), "Volume"),
+    rParamF(Volume,  rShort("volume"), rDefault(0), rLinear(-60.0f,20.0f), "Volume"),
```
Comment (fundamental) [inline src/Params/SUBnoteParameters.cpp]: Could you add units similar to https://github.com/zynaddsubfx/zynaddsubfx/commit/2c035fd23eb21ddde971daeb56af988841f761ff#diff-c3d6d0e24a7c539711c03581b5444362R59 it should just be rUnit(dB).

**Comment by fundamental** [comment]
Looks like with adding the unit, everything should be good to go (I think). For some reason it looks like the link to the travis-ci build is for a different commit SHA ID. I've been meaning to test the commits locally as a result and as such have ended up delaying things a bit.

**Comment by michiboo** [comment]
Hi I have added rUnit(dB) to the parameter!

**Pull request merged**